src/Abp/Runtime/Caching/TypedCacheWrapper.cs

```diff
@@ -78,6 +79,37 @@ public async Task<TValue[]> GetAsync(TKey[] keys, Func<TKey, Task<TValue>> facto
             return values.Select(value => (TValue)value).ToArray();
         }
 
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+            var found = InternalCache.TryGetValue(key.ToString(), out object objectValue);
+            value = CastOrDefault(objectValue);
+            return found;
+        }
+
+        public async Task<ConditionalValue<TValue>> TryGetValueAsync(TKey key)
+        {
+            var result = await InternalCache.TryGetValueAsync(key.ToString());
+            return new ConditionalValue<TValue>(result.HasValue, CastOrDefault(result.Value));
```

Comment:
Shall we define `protected ConditionalValue<TValue> CreateConditionalValue(ConditionalValue<object>)`?

Reply:
can 👍

Reply:
Not done?

Reply:
done in https://github.com/aspnetboilerplate/aspnetboilerplate/pull/5334/commits/2591daba1a711274e00be6f4ddb2d45fc77c4004